### PR TITLE
Create rejections and tests for Book and Inventory aggregates

### PR DESCRIPTION
**Pull request by PaulAgeyev**
Was necessary to cover aggregates for tests, see issue #40.

- Created business logic for handle rejections for Book and Inventory aggregates

- Created tests for them, and for private constructors for rejections classes

- Removed `LocalInMemoryServer` from `todo-list` template

- Also code coverage increased from 75% to 82.75%

### CODE REVIEW

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/BorrowBookCommandTest.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /**
- * @author Dmytry Dyachenko
+ * @author Dmytry Dyachenko, Alexander Karpets
```
Comment (alexanderkarpetc) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/BorrowBookCommandTest.java]: add @author

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/AppendInventoryCommandTest.java [hunk 2]
```diff
@@ -50,6 +50,7 @@ public void setUp() {
     }
 
     @Test
+    @DisplayName("append inventory produce invent")
```
Comment (Dmytry-Dyachenko) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/AppendInventoryCommandTest.java]: What does it mean?

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ExtendLoanPeriodCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        Inventory state2 = aggregate.getState();
+
+        assertEquals(oldDueDate + 60 * 60 * 24 * 14, state2.getLoans(0)
+                                                           .getWhenDue()
+                                                           .getSeconds());
+
+    }
+
+    @Test
+    @DisplayName("not extend loan period")
```
Comment (alexanderkarpetc) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ExtendLoanPeriodCommandTest.java]: repharase

**Review by alexanderkarpetc — CHANGES_REQUESTED** [review]
See my comments + add display name for tests

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/CancelReservationCommandTest.java [hunk 4]
```diff
@@ -58,25 +71,46 @@ void produceEvent() {
...
         dispatchCommand(aggregate, envelopeOf(cancelReservation));
 
         final Inventory inventory = aggregate.getState();
-        assertEquals(0, inventory.getReservationsList().size());
+        assertEquals(0, inventory.getReservationsList()
+                                 .size());
+    }
+
+    @Test
+    @DisplayName("reservation not canceled")
```
Comment (Dmytry-Dyachenko) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/CancelReservationCommandTest.java]: Not consistent with the line 87.
Please, use the same style for `DisplayName`.

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ExtendLoanPeriodCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+public class ExtendLoanPeriodCommandTest extends InventoryCommandTest<ExtendLoanPeriod> {
+
+    @Override
+    @BeforeEach
+    public void setUp() {
+        super.setUp();
+    }
+
+    @Test
+    @DisplayName("extend loan period")
```
Comment (Dmytry-Dyachenko) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ExtendLoanPeriodCommandTest.java]: Add the article.

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ExtendLoanPeriodCommandTest.java (hunk 3 above)
Comment (Dmytry-Dyachenko) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ExtendLoanPeriodCommandTest.java]: The same as the 61 line.

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ReserveBookCommandTest.java [hunk 6]
```diff
@@ -78,5 +94,63 @@ void reserveBook() {
                                                 .getWhoReserved()
                                                 .getEmail()
                                                 .getValue());
+
     }
+
+    @Test
+    @DisplayName("not reserve the book that is already reserved")
```
Comment (Dmytry-Dyachenko) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ReserveBookCommandTest.java]: Maybe you mean: "cannot"?

File: api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ReserveBookCommandTest.java [hunk 7]
```diff
@@ -78,5 +94,63 @@ void reserveBook() {
...
+                                             .getBookId();
+        assertEquals(reserveBook.getInventoryId()
+                                .getBookId(), actualBookId);
+
+        assertEquals(true, rejection.getMessageThrown()
+                                    .getAlreadyReserved());
+    }
+
+    @Test
+    @DisplayName("not reserve the book that is already borrowed")
```
Comment (Dmytry-Dyachenko) [inline api-java/src/test/java/io/spine/javaclasses/exlibris/c/aggregate/definition/ReserveBookCommandTest.java]: The same as the line 101.

**Review by Dmytry-Dyachenko — CHANGES_REQUESTED** [review]
Please, fix all changes and replace `not` with `cannot` in the every rejection test.

**Review by alexanderkarpetc — APPROVED** [review]
LGTM

**Review by Dmytry-Dyachenko — APPROVED** [review]
LGTM

**Pull request merged**